packages/code-studio/src/styleguide/colorUtils.ts

```diff
@@ -149,27 +176,67 @@ export function buildColorGroups(
         acc[group] = [];
       }
 
-      // Skip -hsl variables since they aren't actually colors yet
-      if (/^--dh-color-(.*?)-hsl$/.test(name)) {
-        return acc;
-      }
-
       // Add a spacer for black / white
-      if (name === '--dh-color-black') {
-        acc[group].push({ name: '', value: '' });
+      if (name in SWATCH_LABEL) {
+        acc[group].push({
+          isLabel: true,
+          name: SWATCH_LABEL[name as keyof typeof SWATCH_LABEL],
+          value: '',
+        });
       }
 
       // Skip gray light/mid/dark as we are planning to remove them
       if (/^--dh-color-gray-(light|mid|dark)$/.test(name)) {
         return acc;
       }
 
-      acc[group].push({ name, value });
+      // It might be nice to make these dynamic, but for now just hardcode
+      const note = {
+        '--dh-color-gray-900': 'light',
+        '--dh-color-gray-600': 'mid',
+        '--dh-color-gray-300': 'dark',
+      }[name];
+
+      acc[group].push({ name, value, note });
 
       return acc;
     },
-    {} as Record<string, { name: string; value: string }[]>
+    {} as Record<
+      string,
+      { isLabel?: boolean; name: string; value: string; note?: string }[]
+    >
   );
 
   return groupData;
 }
+
+/**
+ * Normalize color to a hex string and drop alpha if it is 'ff'
+ * @param color
+ */
+export function normalizedOptionalAlpha(color: string): string {
+  return ColorUtils.normalizeCssColor(color).replace(
+    /^(#[a-f0-9]{6})ff$/,
+    '$1'
+  );
+}
+
+export function useContrastFgColorRef<
```

Comment:
I'd prefer this in some sort of hooks file rather than mixed with other utils - this hook requires React and can only be used in a render tree, whereas the other utils do not.